components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/context/SessionContext.java

```diff
@@ -56,6 +55,16 @@ public void setAuthenticatedIdPs(Map<String, AuthenticatedIdPData> authenticated
         this.authenticatedIdPs = authenticatedIdPs;
     }
 
+    public void setAuthenticatedIdPsOfApp(Map<String, Map<String, AuthenticatedIdPData>> authenticatedIdPsOfApp) {
+
+        this.authenticatedIdPsOfApp = authenticatedIdPsOfApp;
+    }
+
+    public Map<String, Map<String, AuthenticatedIdPData>> getAuthenticatedIdPsOfApp() {
```

Comment:
let's not expose the full map here, instead let's have methods to get/set application specific data